tests/test_router_quote.py

```diff
@@ -45,19 +45,8 @@ def test_usdc_in_high_liq_out(self, environment_url, coin_obj):
         amount_str = coin_obj["amount_str"]
         amount_in = int(amount_str)
 
-        # This is the max error tolerance of 7% that we allow.
-        # Arbitrarily hand-picked to avoid flakiness.
-        error_tolerance = 0.07
-        # At a higher amount in, the volatility is much higher, leading to
-        # flakiness. Therefore, we increase the error tolerance to 13%.
-        # The values are arbitrarily hand-picke and can be adjusted if necessary.
-        # This seems to be especially relevant for the Astroport PCL pools.
-        if amount_in > 10_000:
-            error_tolerance = 0.10
-        elif amount_in > 30_000_000_000:
-            error_tolerance = 0.13
-        elif amount_in > 60_000_000_000:
-            error_tolerance = 0.16
+        # Choosse the error tolerance based on amount in swapped.
+        error_tolerance = choose_error_tolerance(amount_in)
```

Comment:
Note: all changes in this test file are drive-by to fix flakiness stemming from high-value amounts. There was a code bug that failed to correctly set the dynamic error tolerance based on the amounts